dubbo-rpc/dubbo-rpc-api/src/main/java/com/alibaba/dubbo/rpc/RpcContext.java

```diff
@@ -610,51 +610,24 @@ public RpcContext setInvocation(Invocation invocation) {
     @SuppressWarnings("unchecked")
     public <T> Future<T> asyncCall(Callable<T> callable) {
         try {
-            try {
-                setAttachment(Constants.ASYNC_KEY, Boolean.TRUE.toString());
-                final T o = callable.call();
-                //local invoke will return directly
-                if (o != null) {
-                    FutureTask<T> f = new FutureTask<T>(new Callable<T>() {
-                        public T call() throws Exception {
-                            return o;
-                        }
-                    });
-                    f.run();
-                    return f;
-                } else {
-
-                }
-            } catch (Exception e) {
-                throw new RpcException(e);
-            } finally {
-                removeAttachment(Constants.ASYNC_KEY);
+            setAttachment(Constants.ASYNC_KEY, Boolean.TRUE.toString());
+            final T o = callable.call();
+            //local invoke will return directly
+            if (o != null) {
+                FutureTask<T> f = new FutureTask<T>(new Callable<T>() {
+                    public T call() throws Exception {
+                        return o;
+                    }
+                });
+                f.run();
+                return f;
             }
-        } catch (final RpcException e) {

```

Comment:
I think we should keep this catch clause, so we can pretend to be a real async call and throw exception with `future.get`

Reply:
can we make a testcase to trigger the second catch statement?

Reply:
Let's wait for @zhaojigang's reply, i think it would be great for himself to add these changes.